architecture/azure/processes.adoc

```diff
@@ -0,0 +1,65 @@
+[[architecture-azure-processes]]
+= Processes
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+This section presents a summary of how {product-title} specific
+processes work. These descriptions do not cover all edge cases and are
+limited to presenting the expected flow.
+
+== Cluster update
+
+A new release might contain new virtual machine (VM) images or container images.
+This process is done in the background without customer involvement or any
+interruption to customer applications running on the cluster.
+The {product-title} resource provider completes all actions.
+The decision about which components, like nodes or Pods, need updating is based on
+comparing the current hash of a component with the desired hash value.
+
+=== Preparation
+
+After the resource provider switches the cluster into the appropriate
+provisioning state, it performs a back up of etcd. The process continues by
+generating all required configuration and template files and placing them in
+their appropriate storage accounts.
+
+=== Master updates
+
+The master nodes update first. The resource provider checks if the current
+hash of master-000000 matches the desired hash value. If the values match,
+the master remains unchanged. If the values do not match, the resource provider
+updates the master by draining, stopping, and updating it to the desired VM; the
+master then restarts.
+
+The process repeats with the other two masters in order, so that only a single
+master stops at a time.
+
+=== Infra and Compute node updates
+
+Next the infra nodes update. In normal update flow, the resource provider creates
```

Comment:
I'm not certain on the capitalization of infra here - my [quick] search yielded mixed results.

s/ss-infra-*/`ss-infra-*`